Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/device/PowerManagement.kt

```diff
@@ -18,7 +21,12 @@ class PowerManagement @Inject constructor(
     private val powerManager by lazy { context.getSystemService<PowerManager>()!! }
 
     val isIgnoringBatteryOptimizations
-        get() = powerManager.isIgnoringBatteryOptimizations(context.packageName)
+        @SuppressLint("NewApi")
+        get() = if (BuildVersionWrap.hasAPILevel(android.os.Build.VERSION_CODES.M)) {
+            powerManager.isIgnoringBatteryOptimizations(context.packageName)
+        } else {
+            true
+        }
 
     val toBatteryOptimizationSettingsIntent =
         Intent(Settings.ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS)
```

Comment:
Same here I think we should use the SDK version checks directly, and provide check also for `toBatteryOptimizationSettingsIntent `

and remove remove `disableBatteryOptimizationsIntent` since it is not used otherwise it has to have a check also

```kotlin
    val isIgnoringBatteryOptimizations
        get() = if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
            powerManager.isIgnoringBatteryOptimizations(context.packageName)
        } else {
            true
        }

    val toBatteryOptimizationSettingsIntent =
        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
            Intent(Settings.ACTION_IGNORE_BATTERY_OPTIMIZATION_SETTINGS)
        } else {
            Intent() // Need to check if this empty intent is not really triggering anything ?
        }
```

Reply:
We could annotate `toBatteryOptimizationSettingsIntent` as `@RequiresApi(Build.VERSION_CODES.M)`?

Reply:
Empty Intent causes `ActivityNotFoundException: No Activity found to handle Intent {  }`. Therefore adding a check at this place does not make sense, as both intents would lead to a crash on Android < 6 (besides that this method should not be called there)